### DISCUSSION TITLE
Fine-tuning with timestamps

### DISCUSSION THREAD

**Discussion by alankras** [openai/whisper#587] [Q&A]
Hello! I successfully tuned a small model without timestamps. WER decreased from 0.6 to 0.29 on my data. The dataset consist of continuous wav's without <|nospeech|> tokens.

Unfortunately, this model lost the ability to predict timestamps. After adding them to the dataset, WER became around 0.35 and visually the model has become much worth.

Has anyone successfully fine-tuned whisper model with timestamps? 
It seems that the problem is in the format of dataset and I can't find any examples from the original paper. Does it make sense to add the <|nospeech|> token, how often does it make sense to insert timestamp tokens (between each word or between phrases), whether it is necessary to put two identical timestamps in a row (one as the ending of the first phrase, another as the beginning to the next)

**Comment by wwbnjsace** [comment]
> Hello! I successfully tuned a small model without timestamps. WER decreased from 0.6 to 0.29 on my data. The dataset consist of continuous wav's without <|nospeech|> tokens.
> 
> Unfortunately, this model lost the ability to predict timestamps. After adding them to the dataset, WER became around 0.35 and visually the model has become much worth.
> 
> Has anyone successfully fine-tuned whisper model with timestamps? It seems that the problem is in the format of dataset and I can't find any examples from the original paper. Does it make sense to add the <|nospeech|> token, how often does it make sense to insert timestamp tokens (between each word or between phrases), whether it is necessary to put two identical timestamps in a row (one as the ending of the first phrase, another as the beginning to the next)

can yo tell me how to fine tune the whisper ?do yo have any code?

**Comment by melaniezhang** [comment]
sorry this is not an answer, but just +1-ing because i would also like to know...

**Comment by giannhskp** [comment]
I have the same question. I have fine-tuned a model using a custom dataset and the model's performance has increased. Though, it is not capable to detect timestamps during inference (like the original models).

How can the model be fine-tuned in order to keep the ability to detect timestamps, like the original models?
